app/main.py

```diff
@@ -1,3 +1,66 @@
+
+
+class Knight:
+
+    def __init__(self, knight: dict) -> None:
+        self.protection = 0
+        self.name = knight["name"]
+        self.power = knight["power"]
+        self.hp = knight["hp"]
+        self.armour = knight["armour"]
+        self.weapon = knight["weapon"]
+        self.potion = knight["potion"]
```

Comment:
Hardcoding dictionary keys  is generally considered a bad practice, a more  flexible approach would be to pass the values directly to the __init__ method.